common/src/main/scala/net/psforever/objects/equipment/EffectTarget.scala

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2017 PSForever
+package net.psforever.objects.equipment
+
+import net.psforever.objects._
+import net.psforever.objects.ce.DeployableCategory
+import net.psforever.objects.serverobject.turret.FacilityTurret
+
+final case class TargetValidation(category : EffectTarget.Category.Value, test : EffectTarget.Validation.Value)
+
+object EffectTarget {
+  /**
+    * A classification of the target of interactions.
+    * Arbitrary, but useful.
+    */
+  object Category extends Enumeration {
+    val
+    Aircraft,
+    Deployable,
+    Equipment,
+    Player,
+    Turret,
+    Vehicle
+    = Value
+  }
+
+  object Validation {
+    type Value = PlanetSideGameObject => Boolean
+
+    def Invalid(target : PlanetSideGameObject) : Boolean = false
+
+    def Medical(target : PlanetSideGameObject) : Boolean = target match {
+      case p : Player =>
+        p.Health > 0 && (p.Health < p.MaxHealth || p.Armor < p.MaxArmor)
+      case _ =>
+        false
+    }
+
+    def HealthCrystal(target : PlanetSideGameObject) : Boolean = target match {
+      case p : Player =>
+        p.Health > 0 && p.Health < p.MaxHealth
+      case _ =>
+        false
+    }
+
+    def RepairSilo(target : PlanetSideGameObject) : Boolean = target match {
+      case v : Vehicle =>
+        !GlobalDefinitions.isFlightVehicle(v.Definition) && v.Health > 0 && v.Health < v.MaxHealth
```

Comment:
Would these need faction checks, or is that applied elsewhere?

Reply:
The only jammering that is faction-specific is the disruptor mine, yet to be implemented.

In terms of proximity terminals, the faction check should be in the `ProximityTerminalControl` object (though I don't think it is).

Reply:
👍